### PR TITLE
WIP: MGMT-21202: Enable dual-stack

### PR DESCRIPTION
**Pull request by danmanor**
### Overview
This PR enables dual-stack networking support for the image-based-install-operator while simplifying the networking configuration to only support machine networks. Cluster networks and service networks have been removed to focus solely on machine/node networking requirements.

#### 🔧 **API Changes**
- **Removed** `ClusterNetworks` and `ServiceNetworks` fields from `ImageClusterInstallSpec`
- **Removed** `ClusterNetworkEntry` struct definition  
- **Kept** both legacy `MachineNetwork` (single) and new `MachineNetworks` (dual-stack) fields
- Updated CRD generation and deepcopy methods accordingly

#### ✅ **Enhanced BMH Validation Logic**
Implemented comprehensive BareMetalHost machine network validation supporting three scenarios:

1. **No Machine Network**: ✅ Valid (no validation required)
2. **Single-Stack**: Validates that at least one NIC has an IP within the specified machine network
3. **Dual-Stack**: Validates that there's a physical NIC (same MAC) with both IPv4 and IPv6 addresses in their corresponding machine networks

**Dual-Stack Validation Features:**
- Separates networks by IP version (IPv4/IPv6)
- Groups NICs by MAC address (representing same physical interface) 
- Requires both IPv4 and IPv6 networks to be present
- Ensures the same physical NIC has addresses in both network families
- Provides detailed error messages for troubleshooting

#### 📋 **Updated Sample Configuration**
```yaml
machineNetworks:
  - cidr: "192.168.1.0/24"    # IPv4
  - cidr: "2001:db8::/64"     # IPv6
```

#### Single-Stack (IPv4)
```yaml
machineNetworks:
  - cidr: "192.168.1.0/24"
```
✅ **Validates**: Any NIC has an IP in `192.168.1.0/24`

#### Dual-Stack 
```yaml
machineNetworks:
  - cidr: "192.168.1.0/24"    # IPv4 network
  - cidr: "2001:db8::/64"     # IPv6 network  
```
✅ **Validates**: Same physical NIC (MAC) has both:
- IPv4 address in `192.168.1.0/24` 
- IPv6 address in `2001:db8::/64`

### Backward Compatibility
- Legacy `MachineNetwork` field still supported
- `MachineNetworks` field takes precedence when both are specified
- No breaking changes for existing single-stack deployments

### Testing
- ✅ All existing tests pass
- ✅ New validation logic handles edge cases
- ✅ Proper error messages for debugging network issues
---

### CODE REVIEW

File: api/v1alpha1/imageclusterinstall_types.go [hunk 1]
```diff
@@ -162,6 +170,12 @@ type Proxy struct {
 	NoProxy string `json:"noProxy,omitempty"`
 }
 
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
+	// CIDR is the IP block address pool for machines within the cluster.
```
[the remark this answers is not in the document]
Comment (danmanor) [reply api/v1alpha1/imageclusterinstall_types.go:175]: I can change if you prefer, I wanted to be aligned with how it is defined in the Installer / install-config / etc.
